### DISCUSSION TITLE
How to use the lammps shared library

### DISCUSSION THREAD

**Discussion by psn417** [deepmodeling/deepmd-kit#843] [Q&A] [answered]
Hello! I am using  [phonolammps](https://phonolammps.readthedocs.io/en/master/introduction.html) do a phonon calculation. Phonolammps need to install lammps python module and shared library (liblammps.so). 

After I install deepmd kit with conda, I type `import lammps` in python, and there is no lammps package.It seems that I need to build deepmd package for lammps, and build lammps shared library. 

But I found there is a `liblammps.so` in `anaconda3/envs/deepmd/lib`. So I want to know how can I use this `liblammps.so`?

**Comment by njzjz** [comment, marked as answer]
You just need to install this package:
https://anaconda.org/deepmodeling/pylammps-dp

**Reply by psn417** [reply]
thank you! after installing it, I can use pylammps. It seems that I need to make a soft link of `liblammps.so` in `site-packages` , otherwise it can not be found by python.

**Comment by psn417** [comment]
I find a strange problem: first, I use ase to relax the atoms to a pressure, 
```
from ase.io import read, write
from deepmd.calculator import DP
from ase.units import GPa
from ase.optimize import BFGS
from ase.constraints import ExpCellFilter
atoms = read("ice-X.cif")
atoms.calc=DP(model="frozen_model.pb")
ecf = ExpCellFilter(atoms, mask=[1,1,1,0,0,0], scalar_pressure=50*GPa)
dyn = BFGS(ecf)
dyn.run(fmax=0.001)
write("relaxed.vasp", atoms, format="vasp", vasp5=True)
write("temp", atoms, format="lammps-data", specorder=["O", "H"])
print("stress tensor in GPa: ", atoms.get_stress()/GPa)
```

then, I use phonolammps to calculate phonon: 
```
from phonolammps import Phonolammps
from phonopy import Phonopy
# use phlammps to get force constants
list_of_commands = ["units metal",
                    "read_data temp",
                    "mass 1 15.999",
                    "mass 2 1.008",
                    "pair_style deepmd frozen_model.pb",
                    "pair_coeff"]

phlammps = Phonolammps(list_of_commands,
                       supercell_matrix=[[2, 0, 0],
                                         [0, 2, 0],
                                         [0, 0, 2]],)
```

I found that these two parts can not be put within a single python script. Otherwise it will give me a `Segmentation fault`  after ase relax is finished. If I divide it into two python scripts, they can run. Do you know why?

**Reply by xfanak** [reply]
Maybe there are some combability issues. Both ASE and deepmd and phonopy have tons of dependency libraries, sometimes you have to specify the version of the dependent package.  I have encountered similar problem when interfacing ASE with pymatgen, and the problem was solved by rolling back pymatgen to an elder version.
